packages/console/src/ds-components/Select/index.tsx

```diff
@@ -58,6 +67,55 @@ function Select<T extends string>({
     event.stopPropagation();
   };
 
+  const getSearchInputContainerStyles = () => {
+    if (!anchorRef.current) {
+      return {};
+    }
+    const element = anchorRef.current;
+    const cs = getComputedStyle(element);
+    const paddingX = Number.parseFloat(cs.paddingLeft) + Number.parseFloat(cs.paddingRight);
+    const paddingY = Number.parseFloat(cs.paddingTop) + Number.parseFloat(cs.paddingBottom);
+    const borderX = Number.parseFloat(cs.borderLeftWidth) + Number.parseFloat(cs.borderRightWidth);
+    const borderY = Number.parseFloat(cs.borderTopWidth) + Number.parseFloat(cs.borderBottomWidth);
+    return {
+      position: 'fixed',
+      width: `${element.offsetWidth - paddingX - borderX}px`,
+      height: `${element.offsetHeight - paddingY - borderY}px`,
+      top: `${
+        element.getBoundingClientRect().top +
+        Number.parseFloat(cs.borderTopWidth) +
+        Number.parseFloat(cs.paddingTop)
+      }px`,
+      left: `${
+        element.getBoundingClientRect().left +
+        Number.parseFloat(cs.borderLeftWidth) +
+        Number.parseFloat(cs.paddingLeft)
+      }px`,
+      backgroundColor: cs.backgroundColor,
+    };
+  };
+
+  useWindowResize(() => {
+    setIsOpen(false);
+  });
+
+  useEffect(() => {
+    setFilteredOptions(
+      searchInputValue
+        ? options.filter(({ value }) =>
+            value.toLocaleLowerCase().includes(searchInputValue.toLocaleLowerCase())
+          )
+        : options
+    );
+  }, [searchInputValue]);
+
+  useEffect(() => {
+    if (isOpen) {
+      anchorRef.current?.scrollIntoView({ block: 'nearest' });
```

Comment:
Is this `scrollIntoView` necessary? I thought when the dropdown is open, the anchor element must have already been in the visible area of the browser window

Reply:
@charIeszhao Yes, because the input element is inside a React modal and always in view, the select box might only be partially visible within the browser window when the dropdown is open.